text/0000-compiler-team-trusted-contrib-maintainer.md

```diff
@@ -0,0 +1,401 @@
+[Code of Conduct][coc] but its spirit.
+
+Team members can choose to take on additional responsibilities, such as
+those listed in the [responsibilities][responsibilities] section. Participating
+in the team's review rotation is encouraged.
+
+If a team member becomes inactive (the contributor's prior contributions and
+other interactions with the project cease) for longer than a year, the team
+member will be moved into alumni status. At any point in the future, they can
+ask to be re-instated at the team member level if they desire.
+
+[Compiler Development Guide]: https://rustc-dev-guide.rust-lang.org/
+
+Maintainers
+-----------
+[maintainers]: #maintainers
+
+Team members are eligible to become maintainers after they have continued to
+contribute actively for a year. Team members can contact team leads or will
+be contacted by team leads to enquire about promotion to maintainership. Team
+members who are eligible for maintainership do not have to become maintainers.
+
+Unlike team members, maintainers are expected to consider themselves
+as exactly that, *maintainers*, of the compiler - put otherwise, to be invested
+in the quality of the compiler codebase and overall health of the compiler
+team, independent of their own projects. Maintainership is primarily intended
+to recognise and encourage participation in activities which are vital to the
+success of the compiler team and broader project.
+
+Maintainers are expected to participate in the ongoing maintenance tasks
+that the compiler team is responsible for (with all of the expected caveats
+for vacation time, mental health breaks, etc) - listed as
+[responsibilities][responsibilities] below. However, not all maintainers need
+to participate in these responsibilities to an equal degree. Maintainers should
+participate in these tasks to the degree that they are able - volunteers are not
+expected to participate as much as those employed to work on the compiler, for
+example. It is the responsibility of the compiler team leads to ensure that the
+ongoing maintenance tasks of the team can be completed sustainably.
+
+Maintainers have all of the same permissions and access as team members.
+Like team members, maintainers are considered members of the Rust project as a
+whole and are expected to follow the spirit of the [Code of Conduct][coc].
+
+Like team members, after inactivity for longer than a year, a
+maintainer will be moved to alumni status. Maintainers who are no longer able to
+help maintain the compiler but otherwise wish to continue contributing to the
+compiler can also be moved to alumni status and retain their team member
+status. Alumni can ask to be reinstated in the future.
+
+[coc]: https://www.rust-lang.org/policies/code-of-conduct
+
+Responsibilities
+----------------
+[responsibilities]: #responsibilities
+
+There are various responsibilities that a maintainer could take on to help the
+team. All maintainers must participate in at least one activity.
```

Comment:
Do you mean in the short term ("I'm currently sick"/"away on vacation"/"taking a temporary leave"/etc) or something else?

Reply:
Short term (things like you mentioned), but also long term - people who's primary job is do work on Rust should probably not just be capping themselves at one activity just because it's "the requirement". (But this is also where intangible things come into play; I can imagine that someone might be involved in FCPs but also mentoring someone, for example.)

Reply:
Ah, thanks for elaborating! Yes, I would hope that people don't feel they can *only* do one thing but at the same time, we should be very cautious about encouraging folks to overextend themselves.

Reply:
Yeah, it's a tough line to walk. It's maybe hard to put into the right words, but personally, I think maintainers should be thinking about more than just some subset of compiler or really just the code itself. I think maintainership implies that you're thinking about the _Project_, which includes present *and* future.

Reply:
I've added some wording based on what you've discussed here. Let me know if it addresses what you were intending to convey.

Reply:
Looks good!